cpp/src/io/json/json_gpu.cu

```diff
@@ -255,13 +255,16 @@ __global__ void convert_data_to_columns_kernel(parse_options_view opts,
     auto const desc =
       next_field_descriptor(current, row_data_range.second, opts, input_field_index, col_map);
     auto const value_len = static_cast<size_t>(std::max(desc.value_end - desc.value_begin, 0L));
+    const bool is_quoted =
+      *(desc.value_begin - 1) == opts.quotechar && *desc.value_end == opts.quotechar;
```

Comment:
Is it always safe to dereference `*(desc.value_begin - 1)`? I didn't fully grasp `next_field_descriptor` on a quick glance, but I don't see why the value's start couldn't be at the beginning of the input.

Reply:
`next_field_descriptor` strips off quotes and whitespaces.
I had the same doubt as well. I kept this because I saw a similar line at 
https://github.com/karthikeyann/cudf/blob/f38240316bcea8a40e0c160c47a82b1d64556479/cpp/src/io/json/json_gpu.cu#L348

You are right. It isn't safe to deference it, so, I will fix both issues in this PR. (Already this PR got multiple minor bug fixes!)

Reply:
fixed.

Reply:
FYI: It is safe for results of `next_field_descriptor` because it will atleast have `:` as the previous character.